jetty-util/src/main/java/org/eclipse/jetty/util/DateCache.java

```diff
@@ -152,92 +156,91 @@ public TimeZone getTimeZone()
     {
-        long seconds = inDate.getTime() / 1000;
-
-        Tick tick = _tick;
-
-        // Is this the cached time
-        if (tick == null || seconds != tick._seconds)
-        {
-            return ZonedDateTime.ofInstant(inDate.toInstant(), _zoneId).format(_tzFormat);
-        }
-
-        return tick._string;
+        return format(inDate.getTime());
     }
 
     /**
      * Format a date according to our stored formatter.
-     * If it happens to be in the same second as the last formatNow
-     * call, then the format is reused.
+     * If it happens to be in the same second as the last
+     * formatNow call, then the format is reused.
      *
-     * @param inDate the date in milliseconds since unix epoch
-     * @return Formatted date
+     * @param inDate the date in milliseconds since unix epoch.
+     * @return Formatted date.
      */
     public String format(long inDate)
     {
-        long seconds = inDate / 1000;
-
-        Tick tick = _tick;
-
-        // Is this the cached time
-        if (tick == null || seconds != tick._seconds)
-        {
-            // It's a cache miss
-            return ZonedDateTime.ofInstant(Instant.ofEpochMilli(inDate), _zoneId).format(_tzFormat);
-        }
+        return formatTick(inDate).getString(inDate);
+    }
 
-        return tick._string;
+    /**
+     * Format a date according to supplied formatter.
+     *
+     * @param inDate the date in milliseconds since unix epoch.
+     * @return Formatted date.
+     */
+    protected String doFormat(long inDate, DateTimeFormatter formatter)
+    {
+        if (formatter == null)
+            return null;
+        return formatter.format(Instant.ofEpochMilli(inDate));
     }
 
     /**
      * Format a date according to our stored formatter.
      * The passed time is expected to be close to the current time, so it is
      * compared to the last value passed and if it is within the same second,
-     * the format is reused.  Otherwise a new cached format is created.
+     * the format is reused. Otherwise, a new cached format is created.
      *
      * @param now the milliseconds since unix epoch
      * @return Formatted date
+     * @deprecated use {@link #format(long)}
      */
+    @Deprecated
     public String formatNow(long now)
     {
-        long seconds = now / 1000;
-
-        Tick tick = _tick;
-
-        // Is this the cached time
-        if (tick != null && tick._seconds == seconds)
-            return tick._string;
-        return formatTick(now)._string;
+        return format(now);
     }
 
+    @Deprecated
     public String now()
     {
         return formatNow(System.currentTimeMillis());
     }
 
+    @Deprecated
     public Tick tick()
     {
         return formatTick(System.currentTimeMillis());
     }
 
-    protected Tick formatTick(long now)
+    protected Tick formatTick(long inDate)
     {
-        long seconds = now / 1000;
+        long seconds = inDate / 1000;
+
```

Comment:
Let's put a comment here discussing both the reason for 2 ticks and the synchronization strategy:
```suggestion
// Two Ticks are cached so that for monotonically increasing times to not see any jitter from multiple cores.
// The ticks are kept in a volatile field, so there a small risk of inconsequential multiple recalculations
```